error/error.go

```diff
@@ -96,11 +94,19 @@ var (
 	// ErrIsWitness is the error when a request is send to a witness.
 	ErrIsWitness = errors.New("peer is witness")
 	// ErrUnknown is the unknow error.
-	ErrUnknown = errors.New("unknow")
+	ErrUnknown = errors.New("unknown")
 	// ErrResultUndetermined is the error when execution result is unknown.
 	ErrResultUndetermined = errors.New("execution result undetermined")
 )
 
+type ErrQueryInterrupted struct {
+	Signal uint32
+} 
```

Comment:
Is the `Signal` one of some limited values? Can we consider adding some different error values for them? Or, can we add a new type `ErrQueryInterruptedWithSignal` for representing interrupted error with signal code, while keeping the previous `ErrQueryInterrupted` error value to avoid the compatibility problem?

Reply:
> Is the Signal one of some limited values? Can we consider adding some different error values for them?

It's not limited by client-go. TiDB can sends whatever it likes..

> Or, can we add a new type ErrQueryInterruptedWithSignal for representing interrupted error with signal code, while keeping the previous ErrQueryInterrupted error value to avoid the compatibility problem?

Yeah... Painful